cmd/rofl/build/artifacts.go

```diff
@@ -576,8 +615,27 @@ func convertToQcow2(buildEnv env.ExecEnv, fn string) error {
 		return fmt.Errorf("%w\n%s", err, out.String())
 	}
 
+	// Resize to ensure virtual size matches original (prevents truncation).
```

Comment:
Was this also a problem before?

Reply:
I figured out what the issue was. It wasn't related to the environment. The problem was that the padding was done on the host (not in the builder container), and over the bind mount the container saw only the pre‑pad size. Doing either `veritysetup verify` or resizing the image fixed the issue (probably just due to touching it?). So I moved the padding to be run in the builder env and it fixes the issue. I have kept the `veritysetup verify` step, since it's useful as a sanity check, but have removed the resize since it's not needed.